### ISSUE TITLE
Remove Node.js dependency from the build system and upgrade to a recent V8 version

### ISSUE THREAD

**Issue by ahmadov** [eclipsesource/J2V8#477]
The `J2V8` project currently has a dependency on the old `Node.js` (7.x) build system to build and produce all necessary `V8` static libraries. In recent years, the upstream `V8` build system [has been improved](https://v8.dev/docs/embed) and we could use it directly instead of using the Node.js build system.

Switching from `Node.js` to a recent upstream `V8` provides the possibility to simplify the build as well as moving to `J2V8` to `V8 7.x` code base at the same time.

**Comment by ahmadov** [comment]
https://github.com/eclipsesource/J2V8/issues/469
https://github.com/eclipsesource/J2V8/issues/436
https://github.com/eclipsesource/J2V8/issues/390

**Comment by ahmadov** [comment]
eclipsesource/J2V8#469
eclipsesource/J2V8#436
eclipsesource/J2V8#390

**Comment by ahmadov** [comment]
eclipsesource/J2V8#469

**Comment by michaeltheory** [comment]
This is something that is quite important to my organization and what appears to be many other individuals posting issues on this repo over the past few years.

I wonder, is there a way to sponsor this project as I've seen popping up on other GH repos, in order to facilitate more activity like this?

**Comment by irbull** [comment]
Upgrading to a newer V8 will mean the existing Debug API won't work. There is a new API (see https://github.com/eclipsesource/J2V8/issues/478). It's not clear if the APIs are compatible. 

My suggestion would be to move to the new version of V8 and update the J2V8 Major version (maybe even move to 7 to indicate V8 version 7). Then add the debug API as a second step. I think the new version is more important to most people, and the debug can be looked at separately.

**Comment by ahmadov** [comment]
I enabled Github Actions to build J2V8. If you follow the commit above you should find green check mark next to issue's title. This leads you to the build output. There you can find the build the artifact (zip file containing .jar file) in the `Artifacts` dropdown box in the upper right corner.

**Comment by Cretezy** [comment]
Are Windows/Mac builds possible to be added to Actions?

Also, where are the latest current builds published? I only see 4.6/4.8 available in Maven central for the platforms.

**Comment by ahmadov** [comment]
The builds are currently not published to Maven central. The publishing stage can be added later.

@irbull decided to reduce the scope of the J2V8 project, you can find more information here: eclipsesource/j2v8#441. Currently, the Windows/Mac builds are not a priority, but it'd be easy to build with Github Actions. Adding the Windows/Mac builds back using Github Actions might be a good community contribution.

**Comment by Cretezy** [comment]
How well does the new V8 work with J2V8? Is it beta-ready? I'm currently not running production software with this and am ready to build myself.

Great work!

**Comment by ahmadov** [comment]
Currently, I disabled the `Node.js` and `V8-debug` support (and corresponding unit tests) on the branch [v8-7.4.288-build](/eclipsesource/J2V8/tree/v8-7.4.288-build). Except for these unit tests, all others are passed and green. At EclipseSource we use `J2V8` in our [eclipsesource/tabris-js](/eclipsesource/tabris-js) project but we still haven't tried the J2V8 based on the new V8 with the tabris.js. Only after that we can provide a feedback how stable it is.

If you can share your own experience using the new J2V8 this would be great.

**Comment by Cretezy** [comment]
Building with Docker fails with the above error, and without gives:

```
fatal error: libplatform/libplatform.h: No such file or directory
 #include <libplatform/libplatform.h>
```

**Comment by ahmadov** [comment]
This code works for me:
```java
import com.eclipsesource.v8.V8;

public class Main {    
    public static void main(String[] args) {
        V8.createV8Runtime();
        System.out.println(V8.getV8Version());
    }
}
```
This is how I executed it:
```bash
$ javac -cp .:j2v8_linux_x86_64-7.0.0-SNAPSHOT.jar Main.java && java -cp .:j2v8_linux_x86_64-7.0.0-SNAPSHOT.jar Main
V8 version 7.4.288
```

**Comment by Cretezy** [comment]
Ah I see, wasn't creating the runtime before. `V8 version: 7.4.288` woohoo!

Although I still can't build, and having issues with my fork's Actions for some reason (never starting). I have the Windows/MacOS build code at https://github.com/Cretezy/J2V8/blob/v8-7.4.288-build/.github/workflows/main.yml, but can't test it
